config/timewindow.py

```diff
@@ -0,0 +1,108 @@
+""" A window time frame, which can be relative to sunset and sunrise
+"""
+from datetime import datetime, timedelta
+import logging
+
+from astral import Location
+
+from ml_tools import tools
+
+
+class TimeWindow:
+    def __init__(self, start, end):
+        self.start = start
+        self.end = end
+        self.location = None
+        self.last_sunrise_check = None
+
+    def use_sunrise_sunset(self):
+        return self.start.is_relative or self.end.is_relative
+
+    def inside_window(self):
+        if self.use_sunrise_sunset:
+            self.update_sun_times()
+            if self.end.time < self.start.time:
+                return self.start.is_after() or self.end.is_before()
+        return self.start.is_after() and self.end.is_before()
+
+    def update_sun_times(self):
+        if not self.use_sunrise_sunset:
```

Comment:
This is wrong. It looks like you want to make a function call here (or add `@property` to `use_sunrise_sunset`)